src/common/mention_free_text.py

```diff
@@ -0,0 +1,20 @@
+from discord.ext.commands import UserInputError
+
+
+class NotMentionFreeException(UserInputError):
+    pass
+
+
+class MentionFreeText:
+    @classmethod
+    async def convert(cls, ctx, argument):
```

Comment:
convert method? how did you decide on this approach? (want to learn)

Reply:
It's part of discord.py. We can create custom types that integate tightly with discord.py's argument typing system. All it took to apply this was to specify this class as the argument type, and discord.py takes care of everything else, including calling `convert`.

I suppose it's a simple enough approach, can't think of any big bad negatives to it...

Reply:
Not implying there are negatives. Just hadn't seen that approach to argument checking before and was curious (learning) :)